File: push_sale_mobile-master/push_sale_mobile-master/lib/models/distributor.dart
import 'package:push_sale/models/address.dart';
import 'package:push_sale/models/warehouse.dart';

class Distributor {
  final String id;
  final String name;
  final String code;
  // final Address address;

  List<Warehouse>? warehouses;

  Distributor({
    required this.id,
    required this.name,
    required this.code,
    // required this.address,
  });

  static Distributor fromMap(Map<String, dynamic> value) {
    return Distributor(
      id: value["id"],
      name: value["name"],
      code: value["code"],
      // address: Address.fromMap(value["address"]),
    );
  }
}

File: push_sale-master/app/Models/Distributor.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Laravel\Passport\HasApiTokens;

class Distributor extends Model
{
    use HasApiTokens, HasFactory;
    protected $table = "distributor";


    public $incrementing = false;
    protected $keyType = 'string';

    protected $fillable = [
        "id",
        "code",
        "private",
        "is_active",
        "name",
        "phone",
        "email",
        "contact_name",
        "address_id",
    ];

    protected $hidden = ["address_id", "created_at", "updated_at"];

    public function Warehouses()
    {
        return $this->hasMany(Warehouse::class);
    }

    public function Address()
    {
        return $this->belongsTo(Address::class);
    }
}
